lib/Doctrine/ORM/Persisters/Collection/OneToManyPersister.php

```diff
@@ -38,10 +39,23 @@ class OneToManyPersister extends AbstractCollectionPersister
      */
     public function delete(PersistentCollection $collection)
     {
-        // This can never happen. One to many can only be inverse side.
-        // For owning side one to many, it is required to have a join table,
-        // then classifying it as a ManyToManyPersister.
-        return;
+        // The only valid case here is when you have weak entities. In this
+        // scenario, you have @OneToMany with orphanRemoval=true, and replacing
+        // the entire collection with a new would trigger this operation.
+        $mapping = $collection->getMapping();
+
+        if ( ! $mapping['orphanRemoval']) {
+            // Handling non-orphan removal should never happen, as @OneToMany
+            // can only be inverse side. For owning side one to many, it is
+            // required to have a join table, which would classify as a ManyToManyPersister.
+            return;
+        }
+
+        $targetClass = $this->em->getClassMetadata($mapping['targetEntity']);
+
+        return $targetClass->isInheritanceTypeJoined()
```

Comment:
It cannot as we may not have an initialized collection and the only information we have is the owner's identifiers for foreign key based deletion.
EntityPersisters only delete based on PK.